frontend/packages/dev-console/src/utils/pipeline-plugin-utils.ts

```diff
@@ -0,0 +1,59 @@
+import * as _ from 'lodash';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { PipelineOverviewItem } from '@console/shared';
+import { PipelineRunModel, PipelineModel } from '../models';
+import { Pipeline, PipelineRun } from './pipeline-augment';
+
+// label to get the pipelines
+export const PIPELINE_REF = 'app.openshift.io/pipeline-ref';
+
+export const tknPipelineAndPipelineRunsResources = (namespace: string): FirehoseResource[] => {
+  const resources = [
+    {
+      isList: true,
+      kind: referenceForModel(PipelineRunModel),
+      namespace,
+      prop: 'pipelineRuns',
+      optional: true,
+    },
+    {
+      isList: true,
+      kind: referenceForModel(PipelineModel),
+      namespace,
+      prop: 'pipelines',
+      optional: true,
+    },
+  ];
+  return resources;
+};
+
+const getPipelineRunsForPipeline = (pipeline: Pipeline, props): PipelineRun[] => {
+  if (props && !props.pipelineRuns) return undefined;
+  const pipelineRunsData = props.pipelineRuns.data;
+  const PIPELINE_RUN_LABEL = 'tekton.dev/pipeline';
+  const pipelineName = pipeline.metadata.name;
+  return pipelineRunsData.filter((pr: PipelineRun) => {
+    return (
+      pipelineName ===
+      (_.get(pr, ['spec', 'pipelineRef', 'name'], null) ||
+        _.get(pr, ['metadata', 'labels', PIPELINE_RUN_LABEL], null))
+    );
+  });
+};
+
+export const getPipelinesAndPipelineRunsForResource = (
+  resource: K8sResourceKind,
+  props,
+): { pipelines: PipelineOverviewItem } => {
+  if (props && !props.pipelines) return undefined;
```

Comment:
Same issues here with returning `undefined` and the conditional issue.

Reply:
fixed in https://github.com/openshift/console/pull/3014/commits/61c377e657d2ed8303b93b45260f10904fda47f2